### PR TITLE
Intercom: improve connector permission selection

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

This PR edits the set / get permissions logic of the Intercom connector. Nothing fancy, I just added the possibility to set a parent resource as not selectable so that we can have a top level for Help Center and Conversations.

I could have decided to have this top level resources selected meaning that the user can decide that all conversations are synced, even if they create a new Team on Intercom, but this solution was quicker to implement and I'd rather have admins tick each Team they want to give access to. No problem to change that in the future. 

Screenshot 2024-02-13 at 14 52 54

## Risk

Low, connector unused at the moment. 

## Deploy Plan

Nothing special.

### CODE REVIEW

File: connectors/src/connectors/intercom/lib/conversation_permissions.ts [hunk 1]
```diff
@@ -90,55 +93,81 @@ export async function retrieveIntercomConversationsPermissions({
...
-        dustDocumentId: null,
-        lastUpdatedAt: null,
-      }));
-    } else {
+  if (isReadPermissionsOnly) {
+    if (isRootLevel && teamswithReadPermission.length > 0) {
+      resources.push(rootConversationRessource);
+    }
+    if (parentInternalId === teamsInternalId) {
+      teamswithReadPermission.forEach((team) => {
```
Comment (flvndvd) [inline connectors/src/connectors/intercom/lib/conversation_permissions.ts:128]: `nit`
```suggestion
      teamsWithReadPermission.forEach((team) => {
```

File: connectors/src/connectors/intercom/lib/conversation_permissions.ts [hunk 2]
```diff
@@ -90,55 +93,81 @@ export async function retrieveIntercomConversationsPermissions({
...
           lastUpdatedAt: null,
-        };
+        });
       });
     }
-    resources.sort((a, b) => {
-      return a.title.localeCompare(b.title);
-    });
   }
+  resources.sort((a, b) => {
```
Comment (flvndvd) [inline connectors/src/connectors/intercom/lib/conversation_permissions.ts:168]: This should definitely be the default behavior and be abstracted in the Connector 🙃.

**Review by flvndvd — APPROVED** [review]
LGTM

File: connectors/src/connectors/intercom/lib/conversation_permissions.ts (hunk 1 above)
Comment (PopDaph) [reply connectors/src/connectors/intercom/lib/conversation_permissions.ts]: thanks, I added a commit!

**Pull request merged**